code/modules/xenomorph/xeno_structures.dm

```diff
@@ -452,3 +450,90 @@
 	newshot.permutated += src
 	newshot.def_zone = pick(GLOB.base_miss_chance)
 	newshot.fire_at(hostile, src, null, ammo.max_range, ammo.shell_speed)
+
+/obj/structure/xeno/resin/xeno_turret/proc/turret_message()
+	if(!TIMER_COOLDOWN_CHECK(src, COOLDOWN_XENO_TURRETS_ALERT))
+		associated_hive.xeno_message("Our [name] is attacking a nearby hostile [hostile] at [get_area(hostile)] (X: [hostile.x], Y: [hostile.y]).", "xenoannounce", 5, FALSE, hostile, 'sound/voice/alien_help1.ogg', FALSE, null, /obj/screen/arrow/turret_attacking_arrow)
+		TIMER_COOLDOWN_START(src, COOLDOWN_XENO_TURRETS_ALERT, 50 SECONDS)
+
+//////////////////////
+///// xeno jell turret
+//////////////////////
+
+/obj/structure/xeno/resin/xeno_turret/jelly
+	icon = 'icons/Xeno/jelturret.dmi'
+	icon_state = "jel_turret"
+	name = "Growth jelly turret"
+	desc = "A menacing looking construct of resin, it seems to be alive. It fires larvae growth jelly, useful for breeding dens."
+	///Fire rate of the target in ticks
+	firerate = 20
+
+/obj/structure/xeno/resin/xeno_turret/jelly/Initialize(mapload, hivenumber = XENO_HIVE_NORMAL)
+	.=..()
```

Comment:
Very nitpicky, but could you add some spaces here and on line 478 so that it's easier to read?
```suggestion
	. = ..()
```